### PR TITLE
Fix ray test

### PR DESCRIPTION
**Pull request by TomTranter**

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
Debugging GitHub Actions is weird 😅

Summary -

1. For some reason, ray requires `liionpack` to be installed in the given environment as a library (I have no idea why).
2. Now the workflows install `liionpack` using `python setup.py install` before running the tests.
3. Same for the `conda` environment.
4. Conda doesn't care about resolving conflicts between dependency versions while installing a library from source (`setup.py`).
5. `distributed` requires `dask==2021.10.0` and because of the above point, `dask` had to be pinned everywhere.

**Comment by TomTranter** [comment]
Thanks for your work on this. I understand that when installing dask in conda the distributed package is included. I don't particularly like the idea of pinning dask so would we fix all our problems if we just didn't test the pip install?

**Comment by wigging** [comment]
Do you need to explicitly provide `distributed` for the pip install? Running `python -m pip install "dask[complete]"` should install everything needed for Dask. More info at https://docs.dask.org/en/latest/install.html

**Comment by Saransh-cpp** [comment]
Thanks, @wigging @TomTranter! Using "dask[complete]" worked!

File: environment.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ dependencies:
   - click
   - autograd
   - absl-py
-  - dask
+  - dask[complete]
```
Comment (wigging) [inline environment.yml:30]: Why is `dask[complete]` needed for the conda environment? Just using `dask` in the environment file works fine for me. I think `dask[complete]` is just needed for the pip install.

**Pull request merged**